redis/_parsers/hiredis.py

```diff
@@ -198,10 +198,16 @@ async def read_response(
         if not self._connected:
             raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR) from None
 
-        response = self._reader.gets()
+        if disable_decoding:
+            response = self._reader.gets(False)
+        else:
+            response = self._reader.gets()
```

Comment:
why not `response=self._reader.gets(not disable_decoding)`
or
```python
enable_decoding = not disable decoding
response=self._reader.gets(enable_decoding)
```

Reply:
It's been a while so I don't exactly remember, especially that there wasn't any sophisticated reason.
I think I wanted to make it consistent with the sync version + I had a problem to get to the `gets` definition, and didn't want to take much more time digging deeper.

Reply:
Yes, well, I had a look at the source:
```c
 self->shouldDecode = 1;
    if (!PyArg_ParseTuple(args, "|i", &self->shouldDecode)) {
        return NULL;
    }
```
Basically, a `shouldDecode` argument can just be a Python bool.
so a `should_decode = not disable_decoding` would have been simpler and more readable.  

Anyway, just came across this, don't worry.

Reply:
Thank you for pointing that out, and linking the code! 😄